app/README.md

```diff
@@ -0,0 +1,221 @@
+
+* `-nt` - NoTemp - sets up the test-environment in `./build` and doesn't delete
+it after the test is done. This is useful if you want to inspect the environment
+after the test run. *CAREFUL* because it will *NOT* recompile the binary. You
+will use this flag mostly while developing the `test.sh` file, so that you can
+quickly test if your new tests are correct or not.
+* `-b` - Build - only useful with `-nt` to force a build of the binary. While
+developing the tests, in a first time you might want to use `-nt` until your
+`test.sh` file is OK, then you will work on your go-code. To recompile the go-code,
+use this flag. Once the go-code is OK, you can remove this flag again and work
+on the `test.sh` file.
+
+## Test.sh setup
+
+The common name for using `libtest.sh` is a file called `test.sh` in the
+same directory as the CLI-binary. The usual structure for this file is:
+
+* Global variables and inclusion of `libtest.sh`
+* main-method setting up the environment and starting the tests
+* a list of test-methods
+* helper methods
+* a call to main, so that the main method is at the top and all necessary
+methods are defined before main is called.
+
+### Global Variables and inclusion of `libtest.sh`
+
+| Variable | Default | Explanation |
+| -------- | ------- | ----------- |
+| `DBG_TEST` | `0` | Show the output of the commands: 0=none, 1=test-names, 2=all |
+| `DBG_SRV` | `0` | DBG-level for server, passed as `-debug` to the `conode` command |
+| `NBR` | `3` | Highest number of servers and clients |
+| `NBR_SERVERS` | `NBR` | How many servers to configure |
+| `NBR_SERVERS_GROUP` | `NBR_SERVERS - 1` | How many servers to write to the
+group-file - per default, keep one server out of the group, to test what
+happens if a server gets added later |
+| `APPDIR` | `pwd` | where the `test.sh`-script is located |
+| `APP` | `$(basename $APPDIR)` | The name of the builddir |
+
+All variables can be overriden by defining them _before_ the inclusion of
+`libtest.sh`. The most common setup of a `test.sh` only has the first two
+global variables, `DBG_TEST` and `DBG_SRV`:
+
+```bash
+# Show the output of the commands: 0=none, 1=test-names, 2=all
+DBG_TEST=1
+# DBG-level for call to the app - this is not handled by libtest.sh, but will
+# be used in the helper methods.
+DBG_APP=2
+# DBG-level for server
+DBG_SRV=0
+
+. $(go env GOPATH)/src/github.com/dedis/onet/app/libtest.sh
+```
+
+If a test fails, it is common to change those variables to:
+
+```bash
+DBG_TEST=2
+DBG_SRV=2
+```
+
+So that `libtest.sh` shows the output of the CLI command and the `conode` prints
+at a debug-level of 2.
+
+### `main`
+
+To have the most important method at the top (for easier editing), `main`
+is defined at the beginning. So any edits to tests can be done there.
+
+A typical main looks like this:
+
+```bash
+main(){
+    startTest
+    buildConode
+    test Build
+    test Network
+    stopTest
+}
```

Comment:
This way of writing tests is fallible, because you might forget to include your new test in `main`. Might be better to use a framework for this - see https://github.com/dedis/cothority/pull/1400